documentation/projects/proposals/dark_mode/20240325-implementation_plan_dark_mode.md

```diff
@@ -0,0 +1,263 @@
+# 2024-03-25 Implementation Plan: Dark Mode
+
+**Author**: @zackkrida
+
+<!-- See the implementation plan guide for more information: https://github.com/WordPress/openverse/tree/19791f51c063d0979112f4b9f4eeace04c8cf5ff/docs/projects#implementation-plans-status-in-rfc -->
+<!-- This template is exhaustive and may include sections which aren't relevant to your project. Feel free to remove any sections which would not be useful to have. -->
+
+## Reviewers
+
+<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->
+
+- [ ] @obulat
+- [ ] @sarayourfriend
+
+## Project links
+
+<!-- Enumerate any references to other documents/pages, including milestones and other plans -->
+
+- [Project Thread](https://github.com/WordPress/openverse/issues/3592)
+- [Project Proposal](./20240313-project_proposal_dark_mode.md)
+
+## Overview
+
+<!-- An overview of the implementation plan, if necessary. Save any specific steps for the section(s) below. -->
+
+This dark mode implementation plan is comprised of three work streams:
+
+- Color management in Tailwind and Vue frontend components
+- Toggling dark mode (new UI and dark mode detection logic)
+- Visual regression tests and feature flagging
+
+Most of these work streams can happen in parallel which I will elaborate on in
+the ["Implementation"](#implementation) section.
+
+### Design Philosophy
+
+Understanding the way @fcoveram has designed the color system for dark mode is
+crucial to understanding this implementation. Quite simply and elegantly, the
+designs use a "palette swap" approach in which each color has a 1:1 replacement
+from light mode to dark mode.
+
+![Openverse.org dark mode color palette](/_static/dark_mode_palette_example.png)
+
+While we _will_ include easy mechanisms for exceptions to this rule, they do not
+appear to be necessary based on the designs. Implementing our dark mode, then,
+should allow component authors to write components _once_, using semantic color
+names, that will automatically switch between their light and dark mode
+counterparts.
+
+### Implementation
+
+We will switch our color names defined in the tailwind configuration to use
+semantic names, for example replacing "pink" with "primary" and "yellow" with
+"complementary". Instead of hardcoding these colors in the Tailwind
+configuration, the tailwind configuration will reference CSS variables defined
+in our root css file. The value of the CSS variables will be switched based on a
+dark mode CSS class added to the HTML root when dark mode is enabled.
+
+Tailwind's built in `dark:` modifier can be used for any styles which need to
+override the default behavior or add dark-mode specific styles beyond the core
+palette swap.
```

Comment:
I was surprised that the semantic colour names wasn't Tailwind's built-in dark mode handling!

Here are the Tailwind docs for darkmode, might be worth linking for context and ease of reference: https://tailwindcss.com/docs/dark-mode

Reply:
I'll definitely add this :+1:

Reply:
Added in 6295bb0a496428a19e22329f4dd4e1c863a60f17.